core/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -39,6 +41,8 @@
 
     private final AtomicLong taskIdGenerator = new AtomicLong();
 
+    private final Collection<TaskManagerListener> listeners = new CopyOnWriteArrayList<>();
```

Comment:
spoke to igor and we agreed that we will make it strict until we really need the modifiability here.